core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -272,8 +270,18 @@ class TaskMetrics private[spark] () extends Serializable {
    */
   @transient private[spark] lazy val _externalAccums = new ArrayBuffer[AccumulatorV2[_, _]]
 
-  private[spark] def externalAccums: Seq[AccumulatorV2[_, _]] = withReadLock {
-    _externalAccums.toArray.toImmutableArraySeq
+  private[spark] def withExternalAccums[T](op: ArrayBuffer[AccumulatorV2[_, _]] => T)
```

Comment:
looks like we only need this API?
```
t.withExternalAccums(_.foreach(...))
t.withExternalAccums(_.flatMap(...))
```

Reply:
Maybe also a filter w/o unnecessary Option creations

Reply:
Updated.